charts/templates/statefulset.yaml

```diff
@@ -31,83 +31,23 @@ spec:
               {{- toYaml .Values.extraVolumeMounts | nindent 12 }}
             {{- end }}
           env:
-            - name: ML_ADMIN_USER
+            - name: MARKLOGIC_ADMIN_USERNAME
               valueFrom:
                 secretKeyRef:
                   name: {{ include "marklogic.fullname" . }}
                   key: username
-            - name: ML_ADMIN_PASS
+            - name: MARKLOGIC_ADMIN_PASSWORD
               valueFrom:
                   secretKeyRef:
                     name: {{ include "marklogic.fullname" . }}
                     key: password
+            - name: MARKLOGIC_INIT
+              value: "true"
+            - name: REPLICA_COUNT
+              value: {{ .Values.replicaCount  | quote }}
           envFrom:
             - configMapRef:
                 name: {{ include "marklogic.fullname" . }}
-          command:
```

Comment:
We want to leave the command, and inside the command run the start up script. As we need to do some extra configuration like turn on/off XDQP encryption in the command section for example.

Reply:
As James mentioned earlier in this though the start up script runs by default, are you recommending I add back the code to run start-marklogic.sh and have sudo touch /var/opt/MarkLogic/ready run after that? Currently in the docker image that line runs only if the REPLICA_COUNT variable is set so it will only run on K8S deployments. I understand wanting to keep that out of the docker image entirely I just want to clarify what you are asking for before I make and commit the changes.

Reply:
If we don't provide commands in statefulSet, the startup script runs by default. However, if we provide the commands, whatever inside the commands will replace the entrypoint/cmd provided in Docker(more detail in https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/). I suggest keep the command to run the startup scripts and extra code like ready and XDQP configuration.

Reply:
Changes have been made

Reply:
As I mentioned in my comment above, I don't want anything specific to k8s in the docker scripts. That is why I would not want something like "REPLICA_COUNT" to be a variable. Tracking the state of the startup and init process is not specific to k8s. 

XDQP encryption is a separate concern and I want to talk through the group-level configuration design.